GLMakie/src/GLVisualize/visualize/particles.jl

```diff
@@ -220,16 +282,52 @@ function draw_scatter((marker, position), data)
     rot = get!(data, :rotation, Vec4f(0, 0, 0, 1))
     rot = vec2quaternion(rot)
     delete!(data, :rotation)
+
+    font = get(data, :font, Observable(Makie.defaultfont()))
+
     # Rescale to include glyph padding and shape
     if isa(to_value(marker), Union{AbstractString, Char})
         scale = data[:scale]
-        font = get(data, :font, Observable(Makie.defaultfont()))
         quad_offset = get(data, :quad_offset, Observable(Vec2f(0)))
-
         # The same scaling that needs to be applied to scale also needs to apply
         # to offset.
         data[:quad_offset] = map(rescale_glyph, marker, font, quad_offset)
         data[:scale] = map(rescale_glyph, marker, font, scale)
+    
+    elseif to_value(marker) isa BezierPath
+        scale = map(combine_scales,
+            pop!(data, :scale, Observable(nothing)),
+            pop!(data, :scale_x, Observable(nothing)),
+            pop!(data, :scale_y, Observable(nothing)),
+            pop!(data, :scale_z, Observable(nothing))
+        )
+        offset = Observable(Vec2f(0))
+        data[:quad_offset] = map(offset_bezierpath, marker, scale, offset)
+        data[:scale] = map(rescale_bezierpath, marker, scale)
+
+    elseif to_value(marker) isa AbstractArray
+        scale = map(combine_scales,
+            pop!(data, :scale, Observable(nothing)),
+            pop!(data, :scale_x, Observable(nothing)),
+            pop!(data, :scale_y, Observable(nothing)),
+            pop!(data, :scale_z, Observable(nothing))
+        )
+        offset = Observable(Vec2f(0))
+        _offset(x::Union{AbstractString, Char}, scale, offset) = rescale_glyph(x, font[], offset)
```

Comment:
We should have just one function, that dispatches on all the various combinations of marker, vector of markers, with different element types (Bezier/Glyph) and remove the complexity from the drawing code.
It would also be nice to calculate scale + offset in one function, to only calculate the boundingbox once.